tests/test_api_search.py

```diff
@@ -403,6 +403,43 @@ def test_create_update_from_raise_exception(self, mock_os_makedirs: mock.MagicMo
 
         self.assertEqual(data, 'Server error - could not create directory')
 
+    @mock.patch('api.views.ycSearch.ycSearch.os.makedirs')
+    def test_create_diff_file_raise_exception(self, mock_os_makedirs: mock.MagicMock):
+        """os.mkdirs() method patched to raise OSError exception.
+        Error message is returned when exception was raised.
+        """
+        mock_os_makedirs.side_effect = OSError()
+
+        file1 = 'yang-catalog'
+        revision1 = '2018-04-03'
+        file2 = 'yang-catalog'
+        revision2 = '2017-09-26'
+
+        path = 'api/services/diff-file/file1={}@{}/file2={}@{}'.format(file1, revision1, file2, revision2)
+        result = self.client.get(path)
+        data = result.data.decode()
+
+        self.assertEqual(data, 'Server error - could not create directory')
+
+
+    @mock.patch('api.views.ycSearch.ycSearch.os.makedirs')
+    def test_create_diff_tree_raise_exception(self, mock_os_makedirs: mock.MagicMock):
+        """os.mkdirs() method patched to raise OSError exception.
+        Error message is returned when exception was raised.
+        """
+        mock_os_makedirs.side_effect = OSError()
+
+        file1 = 'yang-catalog'
+        revision1 = '2018-04-03'
+        file2 = 'yang-catalog'
+        revision2 = '2017-09-26'
+
+        path = 'api/services/diff-tree/file1={}@{}/file2={}@{}'.format(file1, revision1, file2, revision2)
+        result = self.client.get(path)
+        data = result.data.decode()
+
+        self.assertEqual(data, 'Server error - could not create directory')
+
```

Comment:
please separate these changes in different PR. You need to create a new branch for yourself and out of that branch create a separate PR.

Reply:
sorry, I was having trouble with git, done should be done now